pandas/core/reshape/concat.py

```diff
@@ -585,6 +624,66 @@ def _maybe_check_integrity(self, concat_index: Index):
                     "{overlap!s}".format(overlap=overlap)
                 )
 
+    def _items_overlap_with_suffix(self, objs, suffixes):
+        """
+        Adding suffix for items if there is overlapping situation.
+
+        Be aware that `objs` can be either DataFrame-like or Index-like given
+        if `self._is_series` is True or False.
+        """
+        if self._is_series:
+
+            # when _is_series is True, objs are actually column Index
+            overlap_cols = list(objs)
+        else:
+            overlap_cols = chain.from_iterable([obj.columns for obj in objs])
+        to_rename = [col for col, cnt in Counter(overlap_cols).items() if cnt > 1]
+
+        if len(to_rename) == 0 or suffixes is None:
+            return objs
+
+        if not isinstance(suffixes, tuple):
+            raise ValueError(
+                "Invalid type {t} is assigned to suffixes, only"
+                " <class 'tuple'> is allowed.".format(t=type(suffixes))
+            )
+
+        if len(objs) != len(suffixes):
+            raise ValueError(
+                "Number of objects for concatenation is not"
+                "equal to number of suffixes"
+            )
+
+        def renamer(x, suffix):
+            """
+            Rename the indices.
+
+            If there is overlap, and suffix is not None, add
+            suffix, otherwise, leave it as-is.
+
+            Parameters
+            ----------
+            x : original column name
+            suffix : str or None
+
+            Returns
+            -------
+            x : renamed column name
+            """
+            if x in to_rename and suffix is not None:
+                return "{x}{suffix}".format(x=x, suffix=suffix)
```

Comment:
NIT: f string?

Reply:
sure, just my curiosity: any difference/compatibility reason that prefers `f string`?